docs/implementation.md

```diff
@@ -0,0 +1,67 @@
+# Implementation
+
+## Goal
+
+1. Online event-based diff test
+2. Decoupled
+3. Serialization
+
```

Comment:
You can add an overview of different CPU tests infrastructural
1. chipsalliance/rocket-chip like Trace IO
2. sequencer/vector like online difftest with grand central DPI
3. FireSim
4. traditional SystemVerilog testbench

Reply:
Maybe just an overview of different co-simulation frameworks?